http-client/src/test/java/io/airlift/http/client/TestRequestBuilder.java

```diff
@@ -56,6 +58,36 @@ public void testBuilderFromRequest()
         assertEquals(fromRequest(request).build(), request);
     }
 
+    @Test
+    public void testAddCaseInsensitive()
+    {
+        Request request = prepareGet()
+                .setUri(URI.create("http://example.com"))
+                .addHeader("my-header", "lower")
+                .addHeader("My-Header", "camel")
+                .build();
+        assertThat(request.getHeaders().keySet()).hasSize(1);
+        assertThat(request.getHeaders().entries())
+                .containsExactlyInAnyOrder(
+                        entry("my-header", "lower"),
+                        entry("my-header", "camel"));
+    }
+
+    @Test
+    public void testSetCaseInsensitive()
+    {
+        Request request = prepareGet()
+                .setUri(URI.create("http://example.com"))
+                .addHeader("my-header", "lower")
+                .addHeader("My-Header", "camel")
```

Comment:
@losipiuk suggested better test cases for https://github.com/airlift/airlift/pull/1165, so please update